model/storage/RdsStorage.php

```diff
@@ -271,6 +271,46 @@ public function getResourcesUriByQuery(string $query, array $options = [], strin
         return $resourcesUri;
     }
 
+    public function getResourcesDataByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL)
+    {
+        $queryBuilder = $this->getQueryBuilder();
+        $queryBuilder->select('rd.'.self::DATA_RESOURCE, 'rd.'.self::DATA_OBJECT);
+        $queryBuilder->from(self::DATA_TABLE_NAME, 'rd');
+
+        $queryBuilder->join('rd', 'statements', 'st',
+            'st.subject = rd.'.self::DATA_RESOURCE);
+
+        $fieldName = self::DATA_OBJECT;
+        $condition = "rd.$fieldName {$this->getLike()} '%$query%'";
+        $queryBuilder->where($condition);
+        $queryBuilder->andWhere(sprintf('rd.%s = \'%s\'', self::DATA_PREDICATE, $predicate));
+
+        if (isset($options['limit'])) {
+            $queryBuilder->setMaxResults((int)$options['limit']);
+        }
+
+        if (isset($options['offset'])) {
+            $queryBuilder->setFirstResult((int)$options['offset']);
+        }
+
+        $sort = isset($options['sort']) ? $options['sort'] : self::DATA_RESOURCE;
+        $order = isset($options['order']) ? strtoupper($options['order']) : ' ASC';
+
+        $queryBuilder->addOrderBy($sort, $order);
+        $queryBuilder->groupBy('rd.'.self::DATA_RESOURCE);
+
+        $result = $this->getPersistence()->query($queryBuilder->getSQL());
+        $resourcesUri = [];
+        /** @var Revision $statement */
+        while ($statement = $result->fetch()) {
+            $resourcesUri[] = [
```

Comment:
Let's improve naming, but you may find something better then in suggestion
```suggestion
            $resourceData[] = [
```